src/sql/workbench/contrib/query/browser/flavorStatus.ts

```diff
@@ -184,26 +193,16 @@ export class ChangeFlavorAction extends Action {
 			return this._showMessage(Severity.Info, nls.localize('alreadyConnected',
 				"A connection using engine {0} exists. To change please disconnect or change connection", currentProvider));
 		}
-		const editorWidget = getCodeEditor(activeEditor);
-		if (!editorWidget) {
-			return this._showMessage(Severity.Info, nls.localize('noEditor', "No text editor active at this time"));
-		}
 
 		// TODO #1334 use connectionManagementService.GetProviderNames here. The challenge is that the credentials provider is returned
 		// so we need a way to filter this using a capabilities check, with isn't yet implemented
-		const ProviderOptions: ISqlProviderEntry[] = [
-			new SqlProviderEntry(mssqlProviderName)
-		];
```

Comment:
There was a TODO here to filter using a capability check. What capability do we need to check?
Right now I added to return all the providers: MSSQL, MSSQL-CMS (not sure what this is?) and PGSQL which doesn't seem right. 
Any details on how we can solve this problem?

Reply:
@kburtram - What to do about this? Not sure how to implement the capabilities check.

Reply:
@swjain23 we should only show the providers that are actually SQL language flavors.  So for example, MSSQL-CMS wouldn't make sense here I think.  The providers need some type of attribute to declare they can serve this function, which is what I think the TODO is all about.  Implementing a generic capabilities feature is a large work item, so it may make more sense to do something that addresses this case specifically.  A challenge is that Connection providers aren't necessarily language flavor providers so this is overloading this method in a way that isn't really intended by it's design.

Reply:
@kburtram - I did add a flag in this iteration but do not think it is the right way as we do set the language provider on connection without any check.

We do this: 
			// TODO make this generic enough to handle non-SQL languages too
			this.doChangeLanguageFlavor(uri, 'sql', connection.providerName);

in sendConnectRequest when user connects a file to a server. 

Also noticed that in our connection drop down we do not show MSSQL-CMS as we filter it like this:


		this._providerTypeSelectBox.setOptions(filteredProviderDisplayNames.filter((providerDisplayName, index) =>
			// Remove duplicate listings (CMS uses the same display name)
			filteredProviderDisplayNames.indexOf(providerDisplayName) === index)
		);

So maybe we can put this at a common place and filter the same way here as well?